proof/invariant-abstract/DetSchedSchedule_AI.thy

```diff
@@ -9732,7 +9640,8 @@ lemma commit_time_active_sc_valid_refills:
       apply (wpsimp wp: assert_inv is_round_robin_wp)+
   apply (clarsimp simp: obj_at_def cur_sc_more_than_ready_def)
   apply (subgoal_tac "cur_sc_active s")
-  apply (fastforce intro: current_time_bounded_strengthen consumed_time_bounded_helper)
+   apply (clarsimp intro: current_time_bounded_strengthen consumed_time_bounded_helper)
```

Comment:
I remember Japheh saying something about an `intro` not actually doing anything in a `clarsimp`. Does this use this rule? I think he said it requires `intro!: ` to use it. Maybe I'm way off here

Reply:
You were right!